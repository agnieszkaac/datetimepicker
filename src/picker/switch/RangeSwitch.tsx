import React from "react";

export type RangeSwitchProps = {
  value: number;
  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
};

export const RangeSwitch: React.FunctionComponent<RangeSwitchProps> = ({
  value,
  onClick,
}) => (
  <button
    data-testid="range-switch"
    className="range-switch"
    value={value}
    onClick={onClick}
  >
    {value > 0 ? <span>&#10095;</span> : <span>&#10094;</span>}
  </button>
);
