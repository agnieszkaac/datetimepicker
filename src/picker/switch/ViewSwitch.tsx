import React from "react";

export type ViewSwitchProps = {
  label: string;
  onClick: () => void;
};

export const ViewSwitch: React.FunctionComponent<ViewSwitchProps> = ({
  label,
  onClick,
}) => (
  <button
    data-testid="view-switch"
    className="view-switch"
    onClick={onClick}
  >
    {label}
  </button>
);
