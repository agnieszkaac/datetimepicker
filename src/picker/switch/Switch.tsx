import React from "react";
import { Moment } from "moment";

import { RangeSwitch } from "./RangeSwitch";
import { ViewSwitch } from "./ViewSwitch";

import { View } from "../utils";
import { getViewLabel } from "../../utils";

import "./Switch.scss";

export type SwitchProps = {
  view: View;
  viewDate: Moment;
  onViewSwitch: () => void;
  onRangeSwitch: (e: React.MouseEvent<HTMLButtonElement>) => void;
}

export const Switch: React.FunctionComponent<SwitchProps> = ({
  view,
  viewDate,
  onViewSwitch,
  onRangeSwitch,
}) => (
  <div className="switch">
    <RangeSwitch value={-1} onClick={onRangeSwitch} />
    <ViewSwitch label={getViewLabel(view, viewDate)} onClick={onViewSwitch} />
    <RangeSwitch value={1} onClick={onRangeSwitch} />
  </div>
);
