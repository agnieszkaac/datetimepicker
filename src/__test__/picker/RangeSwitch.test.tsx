import React from "react";
import { render, fireEvent } from "@testing-library/react";

import { RangeSwitch, RangeSwitchProps } from "../../picker/switch/RangeSwitch";

describe("<RangeSwitch /> should", () => {
  let props: RangeSwitchProps;
  const onClick = jest.fn();
  props = {
    value: -1,
    onClick,
  };

  it("match snapshot", () => {
    const { asFragment } = render(<RangeSwitch {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("by default render left pointing arrow when value negative", () => {
    const { getByTestId } = render(<RangeSwitch {...props} />);
    const button = getByTestId("range-switch");

    expect(button).toHaveTextContent("❮");
  });
  it("by default render right pointing arrow when value positive", () => {
    const { getByTestId } = render(<RangeSwitch {...props} value={1} />);
    const button = getByTestId("range-switch");

    expect(button).toHaveTextContent("❯");
  });
  it("run onClick function only once", () => {
    const { getByTestId } = render(<RangeSwitch {...props} />);
    const button = getByTestId("range-switch");

    fireEvent.click(button);

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
