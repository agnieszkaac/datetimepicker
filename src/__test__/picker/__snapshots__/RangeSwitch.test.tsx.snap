// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RangeSwitch /> should match snapshot 1`] = `
<DocumentFragment>
  <button
    class="range-switch"
    data-testid="range-switch"
    value="-1"
  >
    <span>
      ‚ùÆ
    </span>
  </button>
</DocumentFragment>
`;
